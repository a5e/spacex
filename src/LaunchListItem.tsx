import React from 'react';
import ILaunch from './data/ILaunch';

interface LaunchListItemProps {
    launch: ILaunch
    handleClick: (flightNumber:number) => void
    isSelected: boolean
}

export default function LaunchListItem(props:LaunchListItemProps) {
    let classes = ['mission_list_item']
    if (props.isSelected) {
        classes.push('selected')
    }
    return (
        <li className={classes.join(' ')}
            onClick={()=> {props.handleClick(props.launch.flight_number);}}
            data-flight-number={props.launch.flight_number}>
            <img src={props.launch.mission_patch_small} alt={props.launch.mission_name + " mission's patch" }/> {props.launch.mission_name}
        </li>
    );
}
