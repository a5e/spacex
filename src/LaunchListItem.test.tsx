import React from 'react';
import ReactDOM from 'react-dom';
import LaunchListItem from './LaunchListItem';
import renderer from 'react-test-renderer';

const launch = {
  "flight_number": 82,
  "mission_name": "CRS-18",
  "launch_year": "2019",
  "launch_date_utc": "2019-07-25T22:01:00.000Z",
  "mission_patch_small": "https://i.imgur.com/X2shYek.png"
}

test('LaunchListItem displays a patch and title', () => {
  const component = renderer.create(
    <LaunchListItem 
      launch = {launch}
      isSelected = {false}
      handleClick = {
        () => {}
      }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('LaunchListItem should have selected class if selected', () => {
  const component = renderer.create(
    <LaunchListItem 
      launch = {launch}
      isSelected = {true}
      handleClick = {
        () => {}
      }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
