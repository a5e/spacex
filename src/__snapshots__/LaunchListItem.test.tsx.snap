// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchListItem displays a patch and title 1`] = `
<li
  className="mission_list_item"
  data-flight-number={82}
  onClick={[Function]}
>
  <img
    alt="CRS-18 mission's patch"
    src="https://i.imgur.com/X2shYek.png"
  />
   
  CRS-18
</li>
`;

exports[`LaunchListItem should have selected class if selected 1`] = `
<li
  className="mission_list_item selected"
  data-flight-number={82}
  onClick={[Function]}
>
  <img
    alt="CRS-18 mission's patch"
    src="https://i.imgur.com/X2shYek.png"
  />
   
  CRS-18
</li>
`;
